<!-- Launch Script for testing comms dockside with WHOI Modem
    John P Morrison
    MIT/WHOI Joint Program
    14 JUN 2025
    Modify the args for agent-to-agent level changes
    Modify the params only for day-to-day changes
-->

<launch>
    <!-- Args that we would change at agent-to-agent level-->
    <arg name="agent_name" default="actor_0" doc="format: actor_0 or beacon_0"/>

    <!-- Set the ROSCONSOLE_FORMAT -->
    <env name="ROSCONSOLE_FORMAT0" value="[${severity}] [${time}] [${node}]: ${message}"/>

    <!-- Start the agent node -->
    <group ns="$(arg agent_name)">

        <!-- Launch the MOOS interface bridge -->
        <node name="udp2ros" pkg="spurdog_moos_bridge" type="udp2ros" output="screen" />

        <node name="rosbag" pkg="rosbag" type="record"
            args='-a -o /ros/logs/$(arg agent_name).bag'
            output="screen" />
    </group>
</launch>